rasa/core/lock_store.py

```diff
@@ -329,3 +352,114 @@ def _load_from_module_name_in_endpoint_config(
+        if key_prefix:
+            logger.debug(f"Setting non-default redis key prefix: '{key_prefix}'.")
+            self._set_key_prefix(key_prefix)
+
+        super().__init__()
+
+    def _set_key_prefix(self, key_prefix: Text) -> None:
+        if isinstance(key_prefix, str) and key_prefix.isalnum():
+            self.key_prefix = (
+                key_prefix + ":" + DEFAULT_CONCURRENT_REDIS_LOCK_STORE_KEY_PREFIX
+            )
+        else:
+            logger.warning(
+                f"Omitting provided non-alphanumeric redis key prefix: '{key_prefix}'. "
+                f"Using default '{self.key_prefix}' instead."
+            )
+
+    def get_lock(self, conversation_id: Text) -> Optional[TicketLock]:
+        """Retrieves lock (see parent docstring for more information)."""
+        serialised_lock = self.red.get(self.key_prefix + conversation_id)
+        if serialised_lock:
+            return TicketLock.from_dict(json.loads(serialised_lock))
+
+        return None
+
+    def delete_lock(self, conversation_id: Text) -> None:
+        """Deletes lock for conversation ID."""
+        deletion_successful = self.red.delete(self.key_prefix + conversation_id)
+        self._log_deletion(conversation_id, deletion_successful)
+
+    def save_lock(self, lock: TicketLock) -> None:
+        """Commit individual tickets, last issued ticket number and lock to storage."""
+        for ticket in lock.tickets:
+            serialised_ticket = ticket.dumps()
+            key = self.key_prefix + lock.conversation_id + ":" + str(ticket.number)
+            self.red.set(name=key, value=serialised_ticket)
+
+        last_issued_ticket_number = lock.last_issued
+        last_issued_key = (
+            self.key_prefix + lock.conversation_id + ":" + "last_issued_ticket_number"
+        )
+        self.red.set(name=last_issued_key, value=last_issued_ticket_number)
+
+        self.red.set(self.key_prefix + lock.conversation_id, lock.dumps())
+
+    def increment_ticket_number(self, lock: TicketLock) -> int:
+        """Uses Redis atomic transaction to increment ticket number."""
+        last_issued_key = (
+            self.key_prefix + lock.conversation_id + ":" + "last_issued_ticket_number"
+        )
+        last_issued_ticket_number = self.red.get(last_issued_key)
+
+        lock_last_issued = lock.last_issued
+
+        if last_issued_ticket_number is None:
+            return lock_last_issued + 1
+
+        if lock_last_issued != last_issued_ticket_number:
+            serialised_ticket = self.red.get(
+                self.key_prefix + lock.conversation_id + ":" + str(lock_last_issued)
+            )
+
+            if serialised_ticket is None:
+                return lock_last_issued + 1
+
+            ticket = Ticket.from_dict(data=json.loads(serialised_ticket))
+            self.red.set(name=last_issued_key, value=ticket.number)
+
+        return self.red.incr(name=last_issued_key)
```

Comment:
I've tried this but it doesn't work, because the `redis-py` client command `incr()` we use doesn't specify a default value to set a key before incrementing it, it only allows you to specify `amount` with which to increment.